NineChronicles.Headless/GraphQLService.cs

```diff
@@ -59,15 +81,58 @@ public void ConfigureServices(IServiceCollection services)
 
                 services
                     .AddSingleton<StandaloneSchema>()
-                    .AddGraphQL((provider, options) =>
+                    .AddGraphQL((options, provider) =>
                     {
                         options.EnableMetrics = true;
-                        options.ExposeExceptions = true;
+                        options.UnhandledExceptionDelegate = context =>
+                        {
+                            Log.Error(
+                                context.Exception,
+                                context.ErrorMessage);
+                        };
                     })
                     .AddSystemTextJson()
                     .AddWebSockets()
                     .AddDataLoader()
+                    .AddUserContextBuilder(context =>
+                    {
+                        if (Configuration["secret"] is { } secretToken
+                            && context.Request.Headers.TryGetValue("Authorization", out StringValues v)
+                            && v.Count == 1 && v[0] == $"Basic {secretToken}")
+                        {
+                            context.User.AddIdentity(
+                                new ClaimsIdentity(
+                                    new[]
+                                    {
+                                        new Claim(
+                                            "role",
+                                            "Admin"),
+                                    }));
+                        }
+
+                        return new GraphQLUserContext
+                        {
+                            User = context.User,
+                        };
+                    })
                     .AddGraphTypes(typeof(StandaloneSchema));
+                services.AddHttpContextAccessor()
+                    .AddTransient<IValidationRule, AuthorizationValidationRule>()
+                    .AddSingleton<IAuthorizationEvaluator, AuthorizationEvaluator>()
+                    .AddTransient(
+                        s =>
+                        {
+                            var authSettings = new AuthorizationSettings();
+                            authSettings.AddPolicy(
+                                "LocalPolicy",
```

Comment:
Can we define constant for "LocalPolicy" ?

Reply:
I moved constant strings such you commented, to `const` variables in `GraphQLService` in c06f870. 🙏